Gruntfile.js

```diff
@@ -30,9 +30,13 @@ module.exports = function (grunt) {
         cmd: 'node',
         args: ['tools/validate', ...kAllSuites.map(s => 'src/' + s)],
       },
+      'generate-cache': {
+        cmd: 'node',
+        args: ['tools/gen_cache', 'src/webgpu'],
+      },
       'validate-cache': {
         cmd: 'node',
-        args: ['tools/gen_cache', 'out', 'src/webgpu', '--validate'],
+        args: ['tools/gen_cache', 'src/webgpu', '--validate'],
```

Comment:
If we can make the compression result stable, I would say, we don't need this step, we can instead just regenerate the cache and then check that git is clean at the end of the GitHub Actions check.

Reply:
Is there any existing code in place to check the git status is clean after building?
Note: This also checks for stale output, and gives a clear message on how to fix it.

Reply:
I thought we had something either here or in the spec repo but I'm not sure.
I think all we would need to do is add a step that runs `git diff-files --quiet` which will return an error code > 0 if there are changes to the working tree.
https://stackoverflow.com/questions/2657935/checking-for-a-dirty-index-or-untracked-files-with-git

Regardless this isn't that much code, and having a clear error message is worth it, so never mind.